a3p-integration/proposals/z:acceptance/test-lib/sync-tools.test.js

```diff
@@ -0,0 +1,412 @@
+    },
+  };
+
+  const setResult = newValue => (result = newValue);
+  const query = () => Promise.resolve(result);
+
+  return { setResult, query };
+};
+
+test.serial('wait until contract is deployed', async t => {
+  const { setValue, follow } = makeFakeFollow();
+  const waitP = waitUntilContractDeplyed(
+    'name',
+    {
+      follow,
+      setTimeout: globalThis.setTimeout,
+    },
+    {
+      maxRetries: 5,
+      retryIntervalMs: 1000,
+      log: t.log,
+      errorMessage: 'Contact not deplyed yet',
+    },
+  );
+
+  setTimeout(() => setValue([['name', true]]), 3000); // set desired value after third retry
+
+  await t.notThrowsAsync(waitP);
+});
+
+test.serial('wait until account funded', async t => {
+  const { setResult, query } = makeFakeBalanceQuery();
+
+  const waitP = waitUntilAccountFunded(
+    'agoric12345',
+    { query, setTimeout: globalThis.setTimeout },
+    { denom: 'ufake', value: 100_000 },
+    {
+      maxRetries: 5,
+      retryIntervalMs: 1000,
+      log: t.log,
+      errorMessage: 'Account not funded yet',
+    },
+  );
+
+  const desiredResult = {
+    balances: [
+      {
+        denom: 'ubld',
+        amount: '364095061',
+      },
+      {
+        denom: 'uist',
+        amount: '2257215',
+      },
+      {
+        denom: 'ufake',
+        amount: '100001',
+      },
+    ],
+    pagination: {
+      next_key: null,
+      total: '0',
+    },
+  };
+  setTimeout(() => setResult(desiredResult), 3000); // set desired value after third retry
+  await t.notThrowsAsync(waitP);
+});
+
+test.serial('wait until account funded, insufficient balance', async t => {
+  const { setResult, query } = makeFakeBalanceQuery();
+
+  const waitP = waitUntilAccountFunded(
+    'agoric12345',
+    { query, setTimeout: globalThis.setTimeout },
+    { denom: 'ufake', value: 100_000 },
+    {
+      maxRetries: 5,
+      retryIntervalMs: 1000,
+      log: t.log,
+      errorMessage: 'Account not funded yet',
+    },
+  );
+
+  const desiredResult = {
+    balances: [
+      {
+        denom: 'ubld',
+        amount: '364095061',
+      },
+      {
+        denom: 'uist',
+        amount: '2257215',
+      },
+      {
+        denom: 'ufake',
+        amount: '90000',
+      },
+    ],
+    pagination: {
+      next_key: null,
+      total: '0',
+    },
+  };
+  setTimeout(() => setResult(desiredResult), 3000); // set desired value after third retry
+  await t.throwsAsync(waitP);
+});
+
+test.serial(
+  'wait until offer result, balance update - should throw',
+  async t => {
+    const { setValue, follow } = makeFakeFollow();
+    setValue({ status: {}, updated: 'balance' });
+
+    const waitP = waitUntilOfferResult(
+      'agoric12345',
+      'my-offer',
+      false,
+      { follow, setTimeout: globalThis.setTimeout },
+      {
+        maxRetries: 5,
+        retryIntervalMs: 1000,
+        log: t.log,
+        errorMessage: 'Wrong update type',
+      },
+    );
+
+    await t.throwsAsync(waitP);
```

Comment:
We pretty much always use the form that specifies the error message, so the test is less likely to pass for the wrong reason.
```suggestion
    await t.throwsAsync(waitP, { message: 'some error' });
```

The error message in `throwsAsync` can also be a `/something/` pattern.

Please fix all the `thowsAsycn()` calls.

Reply:
resolved in https://github.com/Agoric/agoric-sdk/pull/10171/commits/57e25e6118e1925f9dcff04147c0bc928257ef36